library/src/test/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapterBaseTest.java

```diff
@@ -37,145 +36,142 @@
 public abstract class HeaderRecyclerViewAdapterBaseTest<VH extends RecyclerView.ViewHolder, H, T>
     extends RobolectricTest {
 
-  private static final int TYPE_ITEM = -1;
-  private static final int TYPE_HEADER = -2;
-  private static final ViewGroup ANY_VIEW_GROUP = mock(ViewGroup.class);
-  private static final int HEADER_POSITION = 0;
-  private static final int ITEM_POSITION = 1;
-
-  protected HeaderRecyclerViewAdapter<VH, H, T> adapter;
-
-  @Before public void setUp() {
-    adapter = getAdapterUnderTest();
-  }
+  public static final int TYPE_ITEM = -1;
+  public static final int TYPE_HEADER = -2;
+  public static final int POSITION_ZERO = 0;
+  public static final int ANY_POSITION = 3;
+  public static final int POSITION_ONE = 1;
+  public static final int HEADER_POSITION = 0;
+  public static final int ANY_NON_HEADER_POSITION = 4;
 
   @Test public void shouldReturnZeroAsItemCountByDefault() {
-    assertEquals(0, adapter.getItemCount());
+    HeaderRecyclerViewAdapter<VH, H, T> defaultAdapter =
+        new HeaderRecyclerViewAdapterBuilder().build();
+
+    assertEquals(0, defaultAdapter.getItemCount());
   }
 
   @Test public void shouldReturnItemCountIfHasNoHeaderConfigured() {
-    List<T> brands = givenAListWithFiveItems();
+    HeaderRecyclerViewAdapter<VH, H, T> adapterWithoutHeaderAndWithFiveItems =
+        new HeaderRecyclerViewAdapterBuilder().withItems(givenAListWithFiveItems()).build();
 
-    adapter.setItems(brands);
-
-    int fiveElements = brands.size();
-    assertEquals(fiveElements, adapter.getItemCount());
+    assertEquals(5, adapterWithoutHeaderAndWithFiveItems.getItemCount());
   }
 
   @Test public void shouldReturnItemCountPlusOneIfHasHeaderConfigured() {
-    List<T> items = givenAListWithFiveItems();
-    H header = givenAHeader();
-
-    adapter.setHeader(header);
-    adapter.setItems(items);
+    HeaderRecyclerViewAdapter<VH, H, T> adapterWithHeaderAndWithFiveItems =
+        new HeaderRecyclerViewAdapterBuilder().withHeader(givenAHeader())
+            .withItems(givenAListWithFiveItems())
+            .build();
 
-    int expectedSize = items.size() + 1;
-    Assert.assertEquals(expectedSize, adapter.getItemCount());
+    Assert.assertEquals(6, adapterWithHeaderAndWithFiveItems.getItemCount());
```

Comment:
No, I don't. I didn't modify that at all. I asked myself the same question.
So, sure, I change it.